lib/msf/core/rpc/json/error.rb

```diff
@@ -11,17 +11,19 @@ module Msf::RPC::JSON
   SERVER_ERROR_MAX = -32000
   SERVER_ERROR_MIN = -32099
   APPLICATION_SERVER_ERROR = -32000
+  MIDDLEWARE_CATCH_INTERNAL_ERROR = -32001
```

Comment:
I'm not a fan of this name, Error are something that _can_ be caught, they don't perform any catching themselves, seems really weird to me

Reply:
It's definitely a hard error to name. We're using it to classify any error we dont catch ourselves, but instead gets caught by the middleware and returned to the end user, hence the weird name.

Reply:
@dwelch-r7, How does `UNHANDLED_INTERNAL_SERVER_ERROR` sound instead?

Reply:
ummm maybe, what's the difference between this error and Application_server_error?

Reply:
Application_server_error is something we've managed to catch with our own error handling. This new error is for exceptions we don't catch ourselves and our Middleware handles for us instead. All we're really doing here is customizing our response

Reply:
To summarise, it wasn't a good idea to add a seperate type of exception just to indicate that it our own middleware caught and returned it, so the above `MIDDLEWARE_CATCH_INTERNAL_ERROR ` has been rolled into `APPLICATION_SERVER_ERROR`. No extensions to `APPLICATION_SERVER_ERROR` were made  `json_rpc_exception_handling.rb` just uses it instead.